### PR TITLE
Beta Login

### PR DESCRIPTION
**Pull request by V1ct0r-SLiNv-Zhu**
+ New login page
+ Beta version everything

Demo is here: http://45.62.98.190/?course=csci1200
- This login thing still need a readme which I will make it up soon.
- There's a problem I haven't fixed, if anyone has some spare time feel free to fix it. The homework server needs to _GET the course info from the URL, for exemple "?course=csci1200", but the login page is separated from index page, the last part of url is passed to the login page(function) and needs to be passed back to the index. 
-  Jesse wants it to be visually more like the default theme on the homework page, if anyone is good at graphic design, go for it.

### CODE REVIEW

**Comment by MasterOdin** [comment]
This is really not worth the effort of potentially using as:
1. It uses md5 for password "safety". If this login script is newer than 5 years, I'd definitely not bother with a login system designed by someone who knows nothing about security
2. If you're going to use a flat-file database (I'd probably suggest SQLite over it), use a txt file and not a doc. Debugging any potential issues in the database should not require opening MS Word to fix (or worry about file formatting nonsense)
3. When you switch over to a txt file, get rid of all the calls to `base64_encode`/`base64_decode` as it's momentarily pointless and will just make things harder to work on for no actual benefit
4. There needs to be a verification process on sign-ups. This is probably a real issue as depending on the machine this runs on, you may or may not be able to send out emails (you'd need to get VCC permission if it's on a department server). Their username should be their RCS ID and then that gets appended to @rpi.edu to form their email address. They should not be allowed to choose their own username/email as the submitted HWs MUST be under their RCS else it will lead to massive headaches for everyone.

Basically, 1 and 4 are really damning on this, especially due to the fact it really segregates the system off RPI's network for no gain except for I guess people with really lousy friends. I'd talk to Labstaff to see if they have a LDAP or something you could authenticate against in PHP over this method. (Mind, this is a useful step for allowing other schools to potentially use it, but for RPI, it should most likely either remain as is or come up with a method of authenticating against LabStaff).

**Comment by V1ct0r-SLiNv-Zhu** [comment]
Well, this is not a perfect solution, and I am not saying it is. I don't think I will suggest to use this login right away anyway. 1.Talking about security, you should just say there is zero security with the original login which the hwserver uses it now. Also, Signup and forget password and other things are deprecated from what you can see, I keep the files there just in case anyone can use them, or not. This is an open-source place, others can also deploy a similar server for some similar uses. It's very easy to remove a functionality than creating one, if you don't like any of this, take it out. Thus, 4. There is no signup process. Writing a script/file/page to generate passwords for students is a way to assign credentials.

On the other hand, if you are so disappointed by this, you can either try to help and improve it, create a new login other than this, or just do nothing. Everything is to be improved, to be perfected, if there's not the first version of anything, there will not be a better version of..."what"? 

Thanks for you suggestion, again, I will be working on it no matter what. Welcome your criticism on my every commit which will makes it better and also welcome to do a new login thing since you have so many great points :)

**Comment by bmcutler** [comment]
Hi Victor & rest-of-team,
I was hoping to have regular meetings this semester to keep track of what you've been up to and suggest needed tasks...  but I've heard nearly nothing (just some minor points from Kiana & Jesse).  Thus I have integrated none of your changes to the real system.  I hope you will give me a status update before you all scatter for the summer.  Or if anyone is around during the summer I will be doing some major changes preparing for Principles of Software in the Fall.
I don't believe its worth your time working on passwords & logins.  We will not be using this at RPI.  Students do not want to remember yet another password and do not need to.  We are hooked in with the RCS authentication system (that piece may not be checked in, and in fact it may not be something you as students can replicate).  I don't claim that its perfect security, but it's certainly sufficient.  
Barb
